Improve SDL rendering

Holladiewal opened the pull request:
By changing the way we pass the framebuffer information to SDL, we can improve performance.

This change should open up a lot of potential for further render improvement.

Some of Video.jakt is written in inline cpp, either because  I couldnt figure out how to use address-of in jakt, or couldn't make cpp happy with the objects  it got, or how to include constants like SDL_TextureAccess_Streaming. This might also be some  limitation in Jakt for  now. 

Side note:  For whatever reason, at least on my laptop, starting jakt_nes via SSH makes the emulator gain a huge chunk of performance, to the point where it might be too fast.  I cannot explain that behaviour, so it should not be relied upon.
But this shows that we might still be held back by (the way we call) SDL2.

Holladiewal committed:
Video+SDL: Improve rendering

Added wrappers for Textures to SDL.jakt
Use Textures in video.jakt to more efficiently pass our internal
framebuffer to SDL.

This might be even more performant if we would store it as [u32] internally as well.

Holladiewal committed:
PPU: Add missing `RGB::to_u32()`

I might have forgotten to include this while salvaging the git branch.
It is here now!

sophiajt commented on src/video.jakt:
```diff
@@ -1,20 +1,36 @@
-        sdl.set_render_draw_color(red: rgb.red, green: rgb.green, blue: rgb.blue, alpha: 255)
+    mut format = 0u32
+    mut access: c_int = 0
+    unsafe {
```
This makes me wonder if we want to extend sdl.jakt to be able to create and draw its own frames.

That way we can keep all the SDL.h stuff in a single file. We may also want to move our RGB palette to being u32 already so we don't have to do the conversion step.

Holladiewal committed:
PPU: Fix find_sprites finding sprites one scanline too early

This also fixes the glitchy line in Super Mario

Holladiewal committed:
Video: Change framebuffer from RGB to u32

Holladiewal committed:
Video+SDL: Change the way we draw to copy the buffer more efficiently

jntrnr commented:
Yeah, looking a bit more I think we should try to keep all the SDL-related stuff in the sdl.jakt file. Can we maybe create a helper function that lives in sdl.jakt that video.jakt can call, so that video.jakt isn't doing anything SDL-specific?

Holladiewal committed:
Video+SDL: Remove SDL dependency from Video.jakt

Thanks to our improvements and some light persuasion of Jakt,
we no longer have needs for the SDL bits in Video.jakt
